src/lib/Simplify.hs

```diff
@@ -438,14 +440,18 @@ simplifyAbs (Abs bs body) = fromPairE <$> do
   substBinders bs \bs' -> do
     ab <- buildScoped $ simplifyBlock body
     refreshAbs ab \decls result -> do
-      getType result >>= isData >>= \case
+      effs <- declNestEffects decls
```

Comment:
Actually, we could probably make this faster: `simplifyBlock` never modifies the set of effects (or effect for that matter) in a block, so we could reuse the annotations of `body`. Now that I look at it, even the `getType result` part is unnecessary, since we could `substM` the type from `body`'s annotation

Reply:
The annotation doesn't always exist, so the `getType result` code path must still be present.  I'm assuming `getType` (of an `Atom`) isn't so expensive that the extra `case` is worth it, but I am removing the `declNestEffects` call.